bgpd/bgp_community.c

```diff
@@ -648,6 +649,33 @@ enum community_token {
 	community_token_unknown
 };
 
+/* Helper to check if a given community is valid */
+static bool community_valid(const char *community)
+{
+	int octets = 0;
+	char **splits;
+	int num;
+
+	frrstr_split(community, ":", &splits, &num);
+
+	for (int i = 0; i < num; i++) {
+		if (strtoul(splits[i], NULL, 10) > UINT16_MAX)
+			return false;
```

Comment:
Won't this leak the data allocated in frrstr_split?

Reply:
Fixed.